src/test/java/com/xiao/users/controller/UserControllerIntegrationTest.java

```diff
@@ -191,4 +191,72 @@ void testFindAllUser_whenEmptyUserList() throws Exception {
                 .andExpect(jsonPath("$.pageable.pageSize", is(10)));
 
     }
+
+    @Test
+    void testUpdateUser_201() throws Exception {
+        User userSaved = userRepository.save(UserUtil.buildUser());
+        UserDto userUpdate = UserUtil.buildUserDto();
+        userUpdate.setUsername(userUpdate.getUsername()+"_updated");
+        userUpdate.setPassword(userUpdate.getPassword()+"_updated");
+        userUpdate.setEmailAddress(userUpdate.getEmailAddress()+"_updated");
+
+        mockMvc.perform(MockMvcRequestBuilders.put("/users/{id}", userSaved.getId())
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(JsonUtil.asJsonString(userUpdate))
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(userSaved.getId()))
+                .andExpect(jsonPath("$.username").value(userUpdate.getUsername()))
+                .andExpect(jsonPath("$.password").value(userUpdate.getPassword()))
+                .andExpect(jsonPath("$.emailAddress").value(userUpdate.getEmailAddress()));
+    }
+
```

Comment:
add one more test case with below scenario:
- userUpdate dto just have username, password and emailAddress is null